### PR TITLE
update distro version

### LINKED ISSUE #82935: Update bundled distro to 1.9.0
### Summary

Update the bundled package [distro](https://pypi.org/pypi/distro) from `1.8.0` to `1.9.0`.

### Issue Type

Bug Report

### Component Name

`lib/ansible/module_utils/distro/__init__.py`

### Ansible Version

2.18

### Configuration

N/A

### OS / Environment

N/A

### Steps to Reproduce

N/A

### Expected Results

N/A

### Actual Results

N/A

### PR DESCRIPTION
**Pull request by DhaneshRaghavan**
##### SUMMARY

Update [distro](https://pypi.org/project/distro/) package version from 1.8.0 to 1.9.0

fixes [#82935](https://github.com/ansible/ansible/issues/82935)

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

Update the version from the dict to use the updated version.

### CODE REVIEW

File: lib/ansible/module_utils/distro/__init__.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 from __future__ import annotations
 
 # The following makes it easier for us to script updates of the bundled code
-_BUNDLED_METADATA = {"pypi_name": "distro", "version": "1.8.0"}
+_BUNDLED_METADATA = {"pypi_name": "distro", "version": "1.9.0"}
```
Comment (Akasurde) [inline lib/ansible/module_utils/distro/__init__.py:25]: @DhaneshRaghavan Thanks for the contribution. There are other changes which goes with this change. 

```diff
diff --git a/lib/ansible/module_utils/distro/_distro.py b/lib/ansible/module_utils/distro/_distro.py
index e57d6b6854..fa03e3da36 100644
--- a/lib/ansible/module_utils/distro/_distro.py
+++ b/lib/ansible/module_utils/distro/_distro.py
@@ -1,4 +1,4 @@
-# Copyright 2015,2016,2017 Nir Cohen
+# Copyright 2015-2021 Nir Cohen
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -30,7 +30,6 @@ Python 2.6 and removed in Python 3.8. Still, there are many cases in which
 access to OS distribution information is needed. See `Python issue 1322
 <https://bugs.python.org/issue1322>`_ for more information.
 """
from __future__ import annotations
 
 import argparse
 import json
@@ -59,7 +58,7 @@ except ImportError:
     # Python 3.7
     TypedDict = dict
 
-__version__ = "1.8.0"
+__version__ = "1.9.0"
 
 
 class VersionDict(TypedDict):
@@ -129,7 +128,9 @@ _DISTRO_RELEASE_BASENAME_PATTERN = re.compile(r"(\w+)[-_](release|version)$")
 # Base file names to be looked up for if _UNIXCONFDIR is not readable.
 _DISTRO_RELEASE_BASENAMES = [
     "SuSE-release",
+    "altlinux-release",
     "arch-release",
+    "armbian-release",
     "base-release",
     "centos-release",
     "fedora-release",
@@ -155,6 +156,8 @@ _DISTRO_RELEASE_IGNORE_BASENAMES = (
     "system-release",
     "plesk-release",
     "iredmail-release",
+    "board-release",
+    "ec2_version",
 )
 
 
@@ -247,6 +250,7 @@ def id() -> str:
     "rocky"         Rocky Linux
     "aix"           AIX
     "guix"          Guix System
+    "altlinux"      ALT Linux
     ==============  =========================================
 
     If you have a need to get distros for reliable IDs added into this set,
@@ -905,6 +909,9 @@ class LinuxDistribution:
         elif self.id() == "debian" or "debian" in self.like().split():
             # On Debian-like, add debian_version file content to candidates list.
             versions.append(self._debian_version)
+            if self._distro_release_info.get("id") == "armbian":
+                # On Armbian, add version from armbian-release file to candidates list.
+                versions.append(self._armbian_version)
         version = ""
         if best:
             # This algorithm uses the last version in priority order that has
@@ -995,10 +1002,10 @@ class LinuxDistribution:
 
         For details, see :func:`distro.info`.
         """
-        return dict(
+        return InfoDict(
             id=self.id(),
             version=self.version(pretty, best),
-            version_parts=dict(
+            version_parts=VersionDict(
                 major=self.major_version(best),
                 minor=self.minor_version(best),
                 build_number=self.build_number(best),
@@ -1225,6 +1232,16 @@ class LinuxDistribution:
         except FileNotFoundError:
             return ""
 
+    @cached_property
+    def _armbian_version(self) -> str:
+        try:
+            with open(
+                os.path.join(self.etc_dir, "armbian-release"), encoding="ascii"
+            ) as fp:
+                return self._parse_os_release_content(fp).get("version", "")
+        except FileNotFoundError:
+            return ""
+
     @staticmethod
     def _parse_uname_content(lines: Sequence[str]) -> Dict[str, str]:
         if not lines:
@@ -1302,6 +1319,14 @@ class LinuxDistribution:
         if match is not None:
             distro_info["id"] = match.group(1)
 
+            # Armbian release files are not standard as they start with a
+            # comment. Manually set name if it has not been inferred.
+            if distro_info["id"] == "armbian" and distro_info.get(
+                "name", ""
+            ).startswith("#"):
+                distro_info["name"] = "Armbian"
+                distro_info["version_id"] = self._armbian_version
+
         # CloudLinux < 7: manually enrich info with proper id.
         if "cloudlinux" in distro_info.get("name", "").lower():
             distro_info["id"] = "cloudlinux"
```

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/distro/__init__.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/module_utils/distro/__init__.py:25]: FYI we are porting a copy of https://github.com/python-distro/distro/blob/master/src/distro/distro.py to [lib/ansible/module_utils/distro/_distro.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/distro/_distro.py)

Comment (DhaneshRaghavan) [reply lib/ansible/module_utils/distro/__init__.py:25]: thanks @Akasurde for the shedding the light. this is my first time in Ansible so a little bit of missing things.

Comment (Akasurde) [reply lib/ansible/module_utils/distro/__init__.py:25]: No worries. I am also here to help.

**Comment by Akasurde** [comment]
Some CI failures are due to a known issue.

**Comment by DhaneshRaghavan** [comment]
> Some CI failures are due to a known issue.

@Akasurde thanks for the heads-up. I was wondering whether its the new code we introduced.

**Comment by Akasurde** [comment]
> The test `ansible-test sanity --test shebang` [[explain](https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/shebang.html)] failed with 1 error:
> 
> ```
> lib/ansible/module_utils/distro/_distro.py:0:0: should not have a shebang
> ```
> 
> [click here for bot help](https://github.com/ansible/ansibotmini#ansibotmini)

@DhaneshRaghavan This error is added by your code.

**Comment by DhaneshRaghavan** [comment]
> > The test `ansible-test sanity --test shebang` [[explain](https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/shebang.html)] failed with 1 error:
> > ```
> > lib/ansible/module_utils/distro/_distro.py:0:0: should not have a shebang
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [click here for bot help](https://github.com/ansible/ansibotmini#ansibotmini)
> 
> @DhaneshRaghavan This error is added by your code.

@Akasurde it should be fixed by now. Waiting for the sanity 1 to be completed.

File: lib/ansible/module_utils/distro/_distro.py [hunk 2]
```diff
@@ -129,7 +129,9 @@ class InfoDict(TypedDict):
 # Base file names to be looked up for if _UNIXCONFDIR is not readable.
 _DISTRO_RELEASE_BASENAMES = [
     "SuSE-release",
+    "altlinux-release",
     "arch-release",
+    "armbian-release",
```
Comment (bcoca) [inline lib/ansible/module_utils/distro/_distro.py]: this does not seem to be in the 1.9.0 distro, where did it come from?

File: lib/ansible/module_utils/distro/_distro.py [hunk 3]
```diff
@@ -905,6 +910,9 @@ def version(self, pretty: bool = False, best: bool = False) -> str:
         elif self.id() == "debian" or "debian" in self.like().split():
             # On Debian-like, add debian_version file content to candidates list.
             versions.append(self._debian_version)
+            if self._distro_release_info.get("id") == "armbian":
+                # On Armbian, add version from armbian-release file to candidates list.
+                versions.append(self._armbian_version)
```
Comment (bcoca) [inline lib/ansible/module_utils/distro/_distro.py]: again, not part of the 1.9 update

File: lib/ansible/module_utils/distro/_distro.py [hunk 4]
```diff
@@ -1225,6 +1233,16 @@ def _debian_version(self) -> str:
...
+    @cached_property
+    def _armbian_version(self) -> str:
+        try:
+            with open(
+                os.path.join(self.etc_dir, "armbian-release"), encoding="ascii"
+            ) as fp:
+                return self._parse_os_release_content(fp).get("version", "")
+        except FileNotFoundError:
+            return ""
+
```
Comment (bcoca) [inline lib/ansible/module_utils/distro/_distro.py]: idem

File: lib/ansible/module_utils/distro/_distro.py [hunk 5]
```diff
@@ -1302,6 +1320,14 @@ def _distro_release_info(self) -> Dict[str, str]:
...
             distro_info["id"] = match.group(1)
 
+            # Armbian release files are not standard as they start with a
+            # comment. Manually set name if it has not been inferred.
+            if distro_info["id"] == "armbian" and distro_info.get(
+                "name", ""
+            ).startswith("#"):
+                distro_info["name"] = "Armbian"
+                distro_info["version_id"] = self._armbian_version
+
```
Comment (bcoca) [inline lib/ansible/module_utils/distro/_distro.py]: and again

**Review by bcoca — CHANGES_REQUESTED** [review]
This is not an update but an update and modification, all the armbian related changes should be their own PR.

File: lib/ansible/module_utils/distro/_distro.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/module_utils/distro/_distro.py]: We keep a copy of distro file from https://github.com/python-distro/distro/blob/master/src/distro/distro.py, this is where armbian changes are coming in.

Comment (bcoca) [reply lib/ansible/module_utils/distro/_distro.py]: but that is not part of the 1.9.0 release, which is what this PR is stated to do

Comment (DhaneshRaghavan) [reply lib/ansible/module_utils/distro/_distro.py]: I think I got the issue, we should not take it from the master but from the tag version 1.9.0. I'll push the changes now.

**Review by bcoca — APPROVED** [review]

**Review by Akasurde — APPROVED** [review]

**Pull request merged**